internal/github/checkUnsigned.go

```diff
@@ -0,0 +1,98 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/chia-network/go-modules/pkg/slogs"
+	"github.com/google/go-github/v60/github"
+
+	"github.com/chia-network/github-bot/internal/config"
+)
+
+// UnsignedPRs holds information about pending PRs
+type UnsignedPRs struct {
+	Repo     string
+	PRNumber int
+	URL      string
+}
+
+// CheckUnsignedCommits will return a list of PR URLs that have not been updated in the last 7 days by internal team members.
+func CheckUnsignedCommits(ctx context.Context, githubClient *github.Client, cfg *config.Config) ([]UnsignedPRs, error) {
+	var unsignedPRs []UnsignedPRs
+	teamMembers, err := GetTeamMemberList(githubClient, cfg.InternalTeam)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fullRepo := range cfg.CheckRepos {
+		slogs.Logr.Info("Checking repository", "repository", fullRepo.Name)
+		parts := strings.Split(fullRepo.Name, "/")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid repository name - must contain owner and repository: %s", fullRepo.Name)
+		}
+		owner, repo := parts[0], parts[1]
+
+		communityPRs, err := FindCommunityPRs(cfg, teamMembers, githubClient, owner, repo, fullRepo.MinimumNumber)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, pr := range communityPRs {
+			repoName := pr.GetBase().GetRepo().GetFullName() // Get the full name of the repository
+			slogs.Logr.Info("Checking if PR has unsigned commits", "PR", pr.GetHTMLURL())
+			unsigned, err := isUnsigned(ctx, githubClient, pr, teamMembers) // Handle both returned values
+			if err != nil {
+				slogs.Logr.Error("Error checking if PR has unsigned commits", "repository", repoName, "error", err)
+				continue // Skip this PR or handle the error appropriately
+			}
+			if unsigned {
+				slogs.Logr.Info("PR has unsigned commits", "PR", pr.GetNumber(), "repository", fullRepo.Name, "user", pr.User.GetLogin(), "created_at", pr.CreatedAt)
+				unsignedPRs = append(unsignedPRs, UnsignedPRs{
+					Repo:     repo,
+					PRNumber: pr.GetNumber(),
+					URL:      pr.GetHTMLURL(),
+				})
+			} else {
+				slogs.Logr.Info("No commits are unsigned",
+					"PR", pr.GetNumber(),
+					"repository", fullRepo.Name)
+			}
+		}
+	}
+
+	return unsignedPRs, nil
+}
+
+// Checks if a PR has unsigned commits based on the last update from team members
+func isUnsigned(ctx context.Context, githubClient *github.Client, pr *github.PullRequest, teamMembers map[string]bool) (bool, error) {
```

Comment:
Nitpick, but `isUnsigned` sounds more like it's checking an individual commit. `hasUnsignedCommits` sounds more like what it does imo

Reply:
fixed